src/Tools/dotnet-symbol/Program.cs

```diff
@@ -277,6 +294,19 @@ private Microsoft.SymbolStore.SymbolStores.SymbolStore BuildSymbolStore()
             return store;
         }
 
+        private async ValueTask<AuthenticationHeaderValue> SymwebAuthenticationFunc(CancellationToken token)
+        {
+            try
+            {
+                AccessToken accessToken = await TokenCredential.GetTokenAsync(new TokenRequestContext(["api://af9e1c69-e5e9-4331-8cc5-cdf93d57bafa/.default"]), token).ConfigureAwait(false);
```

Comment:
Am I right in thinking this gets called on every symbol server request? If so, it can be unnecessarily expensive. Tokens are typically valid for several hours (the expiration timestamp is part of the AccessToken). It's wise to cache the token in memory and only call the underlying TokenCredential if the cached value is close to expiration (within ~5 minutes).
Some implementations (e.g. the Managed Identity token endpoint for Azure VMs and Azure App Service) -- but not all -- have their own token caches which makes subsequent calls cheap, but they still incur a network call (albeit to a localhost service).

Reply:
This might have been my fault - I thought there was caching in Azure.Identity like MSAL with an application does (thought it was since it's backed by the library). The docs at https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential.gettokenasync?view=azure-dotnet specify:

> Acquired tokens are cached by the credential instance. Token lifetime and refreshing is handled automatically. Where possible, reuse credential instances to optimize cache effectiveness.

Reply:
This is why Lee in CLRMD cached the token until it was 2 minutes before the expiration.  Should I add similar logic?

Reply:
~The docs say it's cached - If that's not the case I am surprised (I know CLI creds are). I am not sure if browser creds are.~ DateTime has its own issues with time skew, but looking at the implementation of Azure.Identity this might be important given the cascading nature of creds and the fact some of them start a process.

Reply:
So are we ok? Is there something you want me to do to the new token caching code?

Reply:
And looks like MI token caching isn't supported. I am not too concerned with caches and IAM revocations since the tokens are so short lived for other types. This library isn't something I expect to be used much in the MI context though so it's ok.